src/filters/filter_lua_script.cpp

```diff
@@ -0,0 +1,382 @@
+        }
+
+        // Going through the table and set output channels
+        sol::object outputs = lua["output"];
+        if (outputs.get_type() == sol::type::table) {
+            for(int universe_id = 0; universe_id <= 2 * ((sol::table) outputs).size(); universe_id++){
+                sol::object universe = ((sol::table) outputs)[universe_id];
+                if(auto uptr = dmxfish::io::get_universe(universe_id); uptr != nullptr) {
+                    if (universe.get_type() == sol::type::table) {
+                        for (uint16_t chan = 0; chan < 512; chan++) {
+                            sol::object channel = ((sol::table) universe)[chan];
+                            if (channel.get_type() == sol::type::number) {
+                                uint8_t value = ((sol::table) universe)[chan];
+                                (*uptr)[chan] = value;
+//                                std::cout << "test : " << (int) value << std::endl;
+//                            } else if (channel.get_type() != sol::type::nil) {
+//                                std::cout << " test : is not nil : " << std::endl;
+                            }
+                        }
+                    }
+                }
+//            for (const auto &univ_kvpair: (sol::table) outputs) {
+//                sol::object uni_key = univ_kvpair.first;
+//                sol::object universe = univ_kvpair.second;
+//                if (uni_key.get_type() == sol::type::number) {
+//                    std::cout << "number" << std::endl;
+//                    int i = ((sol::lua_value) uni_key).value();
+//
+//                    std::cout << "b1 " << i << std::endl;
+//                }
+//                if (uni_key.get_type() == sol::type::string) {
+//                    std::cout << "string" << std::endl;
+////                    int str = (uni_key).get<sol::lua_value>();
+////                    std::cout << "b " << str << std::endl;
+//                }
+//                int i = ((sol::lua_value) uni_key).get<int>();
+//                if (universe.get_type() == sol::type::table) {
+//                    if(auto uptr = dmxfish::io::get_universe((sol::lua_value) uni_key); uptr != nullptr) {
+//                        for (const auto &channel_kv_pair: (sol::table) universe) {
+//                            sol::object ch_name = channel_kv_pair.first;
+//                            sol::object ch_value = channel_kv_pair.second;
+//                            (*uptr)[ch_name] = ch_value;
+//                            std::cout << "c " << std::endl;
+//                        }
+//                    } else {
+//                            throw std::invalid_argument("The requested universe with id " + std::to_string(uni_key) +
+//                                                        " does not exist anymore");
+//                    }
+//                }
```

Comment:
these were thoughts about getting the values with the patching informations

Reply:
Well I still think that it would be better to perform that lookup in lua as we then do not have to do it on every update. This being done in Lua would also provide more flexibility (for example as we did at Erstiparty).

Reply:
Why we dont have to do that on every update? Now it has to be done in lua on every update, or do i think wrong?
If we use a struct that lua knows, then lua writes directly in this memory and we have to update each channel as well, but dont have to lookup these values in lua.

Reply:
Yeah, I think that you're right about that. This loop will also do not do anything as long as there was no output written. I would have put it inside a separate method for readability anyway.